pkg/daemon/controller/controller.go

```diff
@@ -357,6 +370,24 @@ func (sc *ServiceController) ensureVarsAreUpToDate(envVars map[string]string) (b
 	return restartRequired, nil
 }
 
+// EnsureEnvVarsAreRemoved ensures that the given environment variables are removed from the instance's Windows registry
+// An instance restart is required after they are removed to ensure all processes pick up the updated values.
+func EnsureEnvVarsAreRemoved(registryKey registry.Key, envVarsToRemove []string) (bool, error) {
+	restartRequired := false
```

Comment:
Theres too many functions being named similar names throughout the PR, its getting a bit confusing 😅 
I think it would be best to create a separate package `pkg/daemon/envvar`, and move all the functions and constants that deal specifically with interacting with the Windows registry/API to alter environment variables there. This will help to create some structure, and prevent controller.go from becoming too bloated with general helper functions.

Reply:
@sebsoto I have no issues with creating another package. However, I want to understand the confusing part here. This is the only function introduced in controller package in this PR.  All the other functions related to env vars are pre-existing only they have been now exported or their params have changed through this PR.
The only other ensureEnvVarsAreRemoved exists in cleanup package as that basically goes through the same logic flow calling the functions in this package.

Reply:
The confusing part is across the different commits in this PR, there are 5 separate changes to the function signatures of `controller.ensureEnvVarsAreRemoved`, `ensureVarsAreUpToDate` , and `cleanup.ensureEnvVarsAreRemoved`. Some of that is churn and its hard to keep track of when reviewing.

Reply:
I have moved envVar related functions into separate package. Please lmk if that helps.